tf/edgeml/trainer/bonsaiTrainer.py

```diff
@@ -389,18 +432,61 @@ def train(self, batchSize, totalEpochs, sess,
 
             # This helps in direct testing instead of extracting the model out
 
-            testAcc, testLoss, regTestLoss = sess.run(
-                [self.accuracy, self.loss, self.regLoss], feed_dict=_feed_dict)
+            testAcc, testLoss, regTestLoss, pred = sess.run(
+                [self.accuracy, self.loss, self.regLoss, self.prediction], feed_dict=_feed_dict)
+
             if ihtDone == 0:
-                maxTestAcc = -10000
-                maxTestAccEpoch = i
-            else:
-                if maxTestAcc <= testAcc:
+                if (self.bonsaiObj.isRegression == False):
+                    maxTestAcc = -10000
                     maxTestAccEpoch = i
+                elif (self.bonsaiObj.isRegression == True):
                     maxTestAcc = testAcc
-                    self.saveParams(currDir)
+                    maxTestAccEpoch = i
+
+            else:
+                if (self.bonsaiObj.isRegression == False):
+                    if maxTestAcc <= testAcc:
+                        maxTestAccEpoch = i
+                        maxTestAcc = testAcc
+                        self.saveParams(currDir)
+                elif (self.bonsaiObj.isRegression == True):
+                    print("Minimum Training MAE : ", np.mean(maxTestAcc))
+                    if maxTestAcc >= testAcc:
+                        # For regression , we're more interested in the minimum MAE.
+                        maxTestAccEpoch = i
+                        maxTestAcc = testAcc
+                        self.saveParams(currDir)
+            if(i == totalEpochs-1):
```

Comment:
Why do we need this conditional? The final epoch doesn't guarantee you the best model. We save the best model based on MAE. So why do we store the predictions of this model?

Reply:
I had been using this for something else. But are the predictions of the best model, required to be saved ?